api/model/src/main/java/org/projectnessie/model/Util.java

```diff
@@ -44,15 +43,70 @@ private Util() {}
   public static final char REF_HASH_SEPARATOR = '@';
 
   /**
-   * Convert from path encoded string to normal string.
+   * Convert from path encoded string to normal string, supports all Nessie Spec versions.
+   *
+   * <p>The {@code encoded} parameter is split at dot ({@code .}) characters.
+   *
+   * <p>Legacy compatibility: a dot ({@code .}) character can be represented using ASCII 31 (0x1F)
+   * or ASCII 0 (0x00).
+   *
+   * <p>Escaping, for compatibility w/ <a
+   * href="https://jakarta.ee/specifications/servlet/6.0/jakarta-servlet-spec-6.0.html#uri-path-canonicalization">Jakarta
+   * Servlet Specification 6, URI Path Canonicalization</a>. This is available with Nessie Spec
+   * version 2.2.0, as advertised via {@link NessieConfiguration#getSpecVersion()}.
```

Comment:
It looks like `/` is not allowed (by default) in the body of path segments: `Rejecting Suspicious Sequences` -> `The encoded "/" character (e.g. /path%2Finfo)` (also `\`).

`ContentKeys` allow those chars, so I think we have to escape them too using custom escaping because that URI-encoding is transparent to the new validation rules.

Reply:
Perhaps also the `#` (fragment) char